forge-gui/res/cardsfolder/upcoming/ambergris_citadel_agent.txt

```diff
@@ -0,0 +1,92 @@
+PT:3/2
+K:Haste
+K:Specialize:3
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigDiscard | TriggerZones$ Battlefield | TriggerDescription$ Whenever CARDNAME attacks, you may discard your hand and draw two cards.
+SVar:TrigDiscard:DB$ Discard | Optional$ True | Mode$ Hand | Defined$ You | SubAbility$ DBDraw
+SVar:DBDraw:DB$ Draw | Defined$ You | NumCards$ 2
+SVar:HasAttackEffect:TRUE
+DeckHints:Ability$Discard
+Oracle:Haste\nSpecialize {3}\nWhenever Ambergris, Citadel Agent attacks, you may discard your hand and draw two cards.
+AlternateMode:Specialize
+
+SPECIALIZE:WHITE
+
+Name:Ambergris, Agent of Law
+ManaCost:2 R W
+Types:Legendary Creature Dwarf Cleric
+PT:4/3
+K:Haste
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigDiscard | OptionalDecider$ You | TriggerZones$ Battlefield | TriggerDescription$ Whenever CARDNAME attacks, you may discard your hand and draw two cards. If you do, other creatures you control get +X/+X until end of turn, where X is the number of cards you've discarded this turn.
+SVar:TrigDiscard:DB$ Discard | Optional$ True |  Mode$ Hand | Defined$ You | SubAbility$ DBDraw
+SVar:DBDraw:DB$ Draw | Defined$ You | NumCards$ 2 | SubAbility$ DBPump
+SVar:DBPump:DB$ PumpAll | ValidCards$ Creature.Other+YouCtrl | NumAtt$ X | NumDef$ X
+SVar:X:PlayerCountPropertyYou$CardsDiscardedThisTurn
+SVar:HasAttackEffect:TRUE
+DeckHints:Ability$Discard
+Oracle:Haste\nWhenever Ambergris, Agent of Law attacks, you may discard your hand and draw two cards. If you do, other creatures you control get +X/+X until end of turn, where X is the number of cards you've discarded this turn.
+
+SPECIALIZE:BLUE
+
+Name:Ambergris, Agent of Progress
+ManaCost:2 U R
+Types:Legendary Creature Dwarf Cleric
+PT:4/3
+K:Haste
+T:Mode$ Attacks | ValidCard$ Card.Self | OptionalDecider$ You | Execute$ TrigDiscard | TriggerZones$ Battlefield | TriggerDescription$ Whenever CARDNAME attacks, you may discard your hand and draw three cards
+SVar:TrigDiscard:DB$ Discard | Mode$ Hand | Defined$ You | SubAbility$ DBDraw
+SVar:DBDraw:DB$ Draw | Defined$ You | NumCards$ 3
+SVar:HasAttackEffect:TRUE
+DeckHints:Ability$Discard
+Oracle:Haste\nWhenever Ambergris, Agent of Law Progress attacks, you may discard your hand and draw three cards. 
+
+SPECIALIZE:BLACK
+
+Name:Ambergris, Agent of Tyranny
+ManaCost:2 B R
+Types:Legendary Creature Dwarf Cleric
+PT:4/3
+K:Haste
+T:Mode$ Attacks | ValidCard$ Card.Self | OptionalDecider$ You | Execute$ TrigDiscard | TriggerZones$ Battlefield | TriggerDescription$ Whenever CARDNAME attacks, you may discard your hand and draw two cards. When you do, target creature an opponent controls gets -X/-X until end of turn, where X is the number of cards you've discarded this turn.
+SVar:TrigDiscard:DB$ Discard | Mode$ Hand | Defined$ You | SubAbility$ DBDraw
+SVar:DBDraw:DB$ Draw | Defined$ You | NumCards$ 2 | SubAbility$ DBPump
+SVar:DBPump:DB$ Pump | ValidCards$ Creature.OppCtrl | NumAtt$ -X | NumDef$ -X | IsCurse$ True
```

Comment:
why does this Pump all? It says target, also need to account for the "when you do" part

Reply:
Unresolved, this still Pumps all but the Oracle says otherwise